airbyte-integrations/connectors/source-iterable/source_iterable/source.py

```diff
@@ -35,7 +35,11 @@ class SourceIterable(AbstractSource):
     def check_connection(self, logger, config) -> Tuple[bool, any]:
         try:
             list_gen = Lists(api_key=config["api_key"]).read_records(sync_mode=SyncMode.full_refresh)
-            next(list_gen)
+            # result can be considered valid if the iterable has items or an empty generator
+            # ref: https://stackoverflow.com/a/21525143/3921672
+            _exhausted  = object()
+            if next(list_gen, _exhausted) is _exhausted:
+                print('Successfully connected to Iterable API. No available lists in the project')
```

Comment:
Please check if a for loop will do.
https://stackoverflow.com/questions/14413969/why-does-next-raise-a-stopiteration-but-for-do-a-normal-return
https://softwareengineering.stackexchange.com/questions/112463/why-do-iterators-in-python-raise-an-exception